lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java

```diff
@@ -191,10 +197,16 @@ private void holdConfigElement(DOMDocument domDocument, List<Diagnostic> diagnos
         tempDiagnosticsList.add(tempDiagnostic);
     }
 
+    /**
+     * Compare the required feature set with included feature set for each config element.
+     * @param diagnosticsList
+     * @param tempDiagnosticsList
+     * @param featureGraph
+     */
     private void validateConfigElements(List<Diagnostic> diagnosticsList, List<Diagnostic> tempDiagnosticsList, FeatureListGraph featureGraph) {
         for (Diagnostic tempDiagnostic : tempDiagnosticsList) {
             String configElement = tempDiagnostic.getSource();
-            Set<String> includedFeaturesCopy = new HashSet<String>(includedFeatures);
+            Set<String> includedFeaturesCopy = (includedFeatures == null) ? new HashSet<String>() : new HashSet<String>(includedFeatures);
```

Comment:
If `includedFeatures` is null, then there is no point in doing the following 5 lines. The `includedFeaturesCopy` will always be empty in that case.